pkg/service/record/record.go

```diff
@@ -146,22 +162,39 @@ func (r *Recorder) Start(ctx context.Context, reRecord bool) error {
 	errGrp.Go(func() error {
 		for testCase := range frames.Incoming {
 			testCase.Curl = pkg.MakeCurlCommand(testCase.HTTPReq)
-			err := r.testDB.InsertTestCase(ctx, testCase, newTestSetID, true)
-			if err != nil {
-				if ctx.Err() == context.Canceled {
-					continue
+			if reRecordCfg.TestSet == "" {
+				err := r.testDB.InsertTestCase(ctx, testCase, newTestSetID, true)
+				if err != nil {
+					if ctx.Err() == context.Canceled {
+						continue
+					}
+					insertTestErrChan <- err
+				} else {
+					testCount++
+					r.telemetry.RecordedTestAndMocks()
 				}
-				insertTestErrChan <- err
 			} else {
-				testCount++
-				r.telemetry.RecordedTestAndMocks()
+				r.logger.Info("🟠 Keploy has re-recorded test case for the user's application.")
 			}
 		}
 		return nil
 	})
 
 	errGrp.Go(func() error {
 		for mock := range frames.Outgoing {
+			// Send a copy to global mock channel for correlation manager if available
+			if r.globalMockCh != nil {
+				currMockID := r.mockDB.GetCurrMockID()
+				// Create a deep copy of the mock to avoid race conditions
+				mockCopy := *mock
+				mockCopy.Name = fmt.Sprintf("%s-%d", "mock", currMockID+1)
```

Comment:
Why `currMockID+1` and why not just currMockID?

Reply:
Because when we store a new mock we increse the mock count by one then store so similarly we will have to pass the mock by increasing count by one.